### PR TITLE
Enable recursive proof generation

### PR DESCRIPTION
**Pull request by bobbinth**
This PR enables options for generating proofs suitable for recursion. The main difference from the default proofs is that for recursive proofs we use RPO as the hash function (instead of BLAKE3). This makes recursive proof generation quite a bit slower. On my machine (Apple M1 Pro), recursive proofs take about 10x more time to generate.

### CODE REVIEW

File: air/src/proof.rs [hunk 1]
```diff
@@ -117,20 +117,44 @@ impl ProofOptions {
...
 
     /// Creates a new preset instance of [ProofOptions] targeting 96-bit security level.
-    pub fn with_96_bit_security() -> Self {
-        let options = WinterProofOptions::new(27, 8, 16, FieldExtension::Quadratic, 8, 255);
-        Self {
-            hash_fn: HashFunction::Blake3_192,
-            options,
+    ///
+    /// If `recursive` flag is set to true, proofs will be generated using an arithmetization-
+    /// friendly hash function (RPO). Such proofs a well-suited for recursive proof verification,
```
Comment (Al-Kindi-0) [inline air/src/proof.rs:122]: nit: a -> are

File: air/src/proof.rs [hunk 2]
```diff
@@ -117,20 +117,44 @@ impl ProofOptions {
...
 
     /// Creates a new preset instance of [ProofOptions] targeting 128-bit security level.
-    pub fn with_128_bit_security() -> Self {
-        let options = WinterProofOptions::new(27, 16, 21, FieldExtension::Cubic, 8, 255);
-        Self {
-            hash_fn: HashFunction::Blake3_256,
-            options,
+    ///
+    /// If `recursive` flag is set to true, proofs will be generated using an arithmetization-
+    /// friendly hash function (RPO). Such proofs a well-suited for recursive proof verification,
```
Comment (Al-Kindi-0) [inline air/src/proof.rs:143]: nit: a -> are

**Review by Al-Kindi-0 — APPROVED** [review]
Looks good to me, just a small nit.
Thank you!

File: air/src/proof.rs (hunk 1 above)
Comment (bobbinth) [reply air/src/proof.rs]: Fixed!

File: air/src/proof.rs (hunk 2 above)
Comment (bobbinth) [reply air/src/proof.rs]: Fixed!

**Pull request merged**